lib/ansible/plugins/callback/json.py

```diff
@@ -101,14 +101,22 @@ def v2_playbook_on_stats(self, stats):
             summary[h] = s
 
         custom_stats = {}
+        custom_stats_global = {}
+
         if self.get_option('show_custom_stats') and stats.custom:
-            custom_stats.update(dict((self._convert_host_to_name(k), v) for k, v in stats.custom.items()))
-            custom_stats.pop('_run', None)
+
+            for k in stats.custom.keys():
+                if k == '_run':
+                    custom_stats_global = stats.custom[k]
+                    continue
+
+                custom_stats.update({self._convert_host_to_name(k): stats.custom[k]})
```

Comment:
I feel like this changes too much, for what could be achieved with less:

```diff
diff --git a/lib/ansible/plugins/callback/json.py b/lib/ansible/plugins/callback/json.py
index 3961a78aab..fd344a0cf0 100644
--- a/lib/ansible/plugins/callback/json.py
+++ b/lib/ansible/plugins/callback/json.py
@@ -101,14 +101,16 @@ class CallbackModule(CallbackBase):
             summary[h] = s
 
         custom_stats = {}
+        global_custom_stats = {}
         if self.get_option('show_custom_stats') and stats.custom:
             custom_stats.update(dict((self._convert_host_to_name(k), v) for k, v in stats.custom.items()))
-            custom_stats.pop('_run', None)
+            global_custom_stats.update(custom_stats.pop('_run', {}))
 
         output = {
             'plays': self.results,
             'stats': summary,
             'custom_stats': custom_stats,
+            'global_custom_stats': global_custom_stats,
         }
 
         self._display.display(json.dumps(output, indent=4, sort_keys=True))
```

Reply:
Your code is realy more lean and clean, thanks for the review, i changed it to your suggestion